dynamiq/nodes/node.py

```diff
@@ -511,7 +512,7 @@ def run(
 
             transformed_input = self.transform_input(input_data=input_data, depends_result=depends_result)
 
-            self.run_on_node_start(config.callbacks, transformed_input, **merged_kwargs)
+            transformed_input |= self.run_on_node_start(config.callbacks, transformed_input, **merged_kwargs)
```

Comment:
The callback shouldn't modify input/output. I such cases, that is some other transformers. We already have different type of transformers.
Callbacks are a way just to call smth without affecting the initial flow and data.

Reply:
Refactored, leaving Node callbacks that does not change anything.